main.cpp

```diff
@@ -0,0 +1,225 @@
+/* ----------------------------------------------------------------------------
+* Copyright &copy; 2015 Randy Tanizawa <rtanizawa@csu.fullerton.edu>
+* Released under the [MIT License] (http://opensource.org/licenses/MIT)
+* ------------------------------------------------------------------------- */
+
+/**
+* A demonstration of recursive and iterative functions.
+*   Including: Greatest Common Denominator
+               Fibonacci Numbers
+               Integers to Roman Numerals
+               Integers to Alphabetized Integers
+               Magic Number Riddle
+*/
+
+#include <iostream>
+#include <cmath>
+#include <string>
+
+using std::cout;
+using std::endl;
+using std::abs;
+using std::string;
+
+int gcd( int x, int y );
+int gcd_iter( int x , int y );
+int fib( int f , int f1 = 1, int f2 = 1);
+int fib_iter( int f );
+string int_to_roman( int r , string rn = "");
+string int_to_word( int w, string nw = "");
+int magic( int m );
+
+int main()
+{
+    cout << gcd(24,600) << endl;
+    cout << gcd_iter(24,600) << endl;
+    cout << fib(11) << endl;
+    cout << fib_iter(11) << endl;
+    cout << int_to_word(18818) << endl;
+    magic(12);
+
+    return 0;
+}
+
+int gcd( int x, int y )
+{
+    x = abs(x);
+    y = abs(y);
+
+    if( x == 0 || y == 0 )
+        return x+y;
+
+    if( x > y )
+        return gcd( x-y , y);
+    else
+        return gcd( x , y-x );
+}
+
+int gcd_iter( int x , int y )
+{
+    x = abs(x);
+    y = abs(y);
+
+    while(x && y)
+        if( x > y )
+            x -= y;
+        else
+            y -= x;
+
+    return x+y;
+}
+
+int fib( int f , int f1, int f2 )
+{
+    if( f <= 2 )
+        return f2;
+    else
+    {
+        int f3;
+        f3 = f1 + f2;
+        f1 = f2;
+        f2 = f3;
+        f--;
```

Comment:
rather than modify all these variables, it would be more typical in functional programming (and thus, in recursive functions in general) to perform these operations inside the recursive call to `fib`